### PR TITLE
Added View for Constraints

### PR DESCRIPTION
**Pull request by oliviia12**
New: The perspective includes the constraint view of the current model. When opening a new model you need to reset the perspective to see the current constraints.
Not supposed to respond to changes yet so do not worry if you change the diagram and the view doesn't respond (will follow soon).

### CODE REVIEW

**Review by Indycate — COMMENTED** [review]
@PaulWestphal looks pretty good

**Comment by PaulWestphal** [comment]
The view doesn't show any constraints, even when resetting the perspective.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FeatureModelUtil.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * TODO This is a util class thats provides some helper methods to get feature model context.
+ *
+ * @author "Rosiak Kamil"
+ */
+public class FeatureModelUtil {
+	/**
+	 * This method returns the active feature model editor if available.
+	 *
+	 * @return
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FeatureModelUtil.java]: Either have a return tag or don't. Don't leave it empty.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FeatureModelUtil.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+package de.ovgu.featureide.fm.ui.utils;
+
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.PlatformUI;
+
+import de.ovgu.featureide.fm.core.base.IFeatureModel;
+import de.ovgu.featureide.fm.ui.editors.FeatureModelEditor;
+
+/**
+ * TODO This is a util class thats provides some helper methods to get feature model context.
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FeatureModelUtil.java]: It's unclear what this TODO is for.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/ConstraintViewController.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import de.ovgu.featureide.fm.core.base.IConstraint;
+import de.ovgu.featureide.fm.core.base.IFeatureModel;
+import de.ovgu.featureide.fm.core.base.event.FeatureIDEEvent;
+import de.ovgu.featureide.fm.core.base.event.IEventListener;
+import de.ovgu.featureide.fm.ui.FMUIPlugin;
+import de.ovgu.featureide.fm.ui.utils.FeatureModelUtil;
+import de.ovgu.featureide.fm.ui.views.constraintview.view.ConstraintView;
+
+/**
+ * TODO description
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/ConstraintViewController.java]: Your final version should not have comments like this.

**Comment by oliviia12** [comment]
Did you close the view before reseting? Then it should work. Otherwise it might be Windows-related.

**Comment by oliviia12** [comment]
@PaulWestphal

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/view/ConstraintView.java [hunk 4]
```diff
@@ -18,85 +19,110 @@
...
+		addTableLayout(viewer);
 		table.setHeaderVisible(true);
 		table.setLinesVisible(true);
-		// define layout for the viewer
+	}
 
+	private void addTableLayout(TableViewer viewer) {
+		final TableLayout layout = new TableLayout();
+		layout.addColumnData(new ColumnWeightData(60, true));
+		layout.addColumnData(new ColumnWeightData(40, 800, true));
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/view/ConstraintView.java]: Use final ints with descriptive names for this.

**Comment by PaulWestphal** [comment]
Oh, that was it.

**Review by PaulWestphal — COMMENTED** [review]
Looks good, just make some small changes and I'll merge it.

**Pull request merged**